frontend/alert/src/components/AutoComplete.js

```diff
@@ -206,14 +241,29 @@ const AutoComplete = () => {
                 autocomplete="off"
                 placeholder="Course"
                 ref={setInputRef}
-                value={value}
-                onChange={({ target: { value: newValue } }) => {
-                    setValue(newValue);
-                }}
-                onKeyDown={(e) => {
-                    // autocomplete with backdrop when the right arrow key is pressed
+                onKeyUp={(e) => {
                     if (e.keyCode === 39 && inputRef && suggestions && suggestions[0]) {
+                        // autocomplete with backdrop when the right arrow key is pressed
                         setValue(backdrop);
+                        inputRef.value = backdrop;
+                    } else if (e.keyCode === 40 && suggestions) {
+                        // select a suggestion when the down arrow key is pressed
+                        const newIndex = getCurrentIndex() + 1;
+                        const newSelectedSuggestion = Math.min(newIndex, suggestions.length - 1);
+                        handleSuggestionSelect(newSelectedSuggestion);
+                    } else if (e.keyCode === 38 && suggestions) {
+                        // select a suggestion when the down arrow key is pressed
+                        const newSelectedSuggestion = Math.max(getCurrentIndex() - 1, -1);
+                        handleSuggestionSelect(newSelectedSuggestion);
+                    } else {
+                        const newValue = e.target.value;
+                        setValue(newValue);
```

Comment:
looks like `value` isn't necessarily going to be the `value` inside of the input element (that we get from `e.target.value`) and will only match up in this else case. This is fine, but I'm pretty sure it's non-standard for handling forms in React -- an explanatory comment where the state is defined would go a long way here, and potentially also renaming `value` to be `currentSearch` or something might also help.